### ISSUE TITLE
get_object performance

### ISSUE THREAD

**Issue by richardrowe** [pboehm/django-data-migration#8]
I am attempting to run several migrations with django-data-migration library. The data set is very large, and the migrations without any depends_on (or usage of is_a) run relatively quickly but when I get to any migration that has an is_a performance drops immediately (x1000~ slower?). I suspect it could be get_object, but wondered if you experienced similar results?

I might swap this out to just meet my needs for performance if you're not experiencing similar results.

**Comment by pboehm** [comment]
I must apologize that I haven't used this on a really large result set (max 30000 entries). 

When where are FK/O2O/M2M relations in your model and also configured in the `column_description` with `is_a`. For every element in the result set there will be queries that will search the related objects that have already been migrated. 

This decreases the performance immensely but I haven't got another solution which also leads to valid model instances. Did you have an idea?

**Comment by richardrowe** [comment]
@pboehm thanks. I ended up using a global dictionary to map out relationships and create them later as an interim solution. :-)

**Comment by aidanlister** [comment]
Yeah -- my migrations take several hours to run now, which is pretty painful when it dies at the last minute.

I guess a hook_prepare_map() which returned a dictionary of dictionaries for each relation would be a decent solution.

The migation would then check if a key exists for that relationship, if it did it would use the hashtable, otherwise it would use the default database lookup.

My hook_prepare_map might look like this (mapping "ref" to the created "pk"):

```
properties = Property.objects.all().values_list('ref', 'pk')
maps['property'] = dict(properties)
```

**Comment by pboehm** [comment]
I will integrate a solution for this problem as soon as possible.

**Comment by pboehm** [comment]
I have published version 0.2.0 which includes some performance improvements like prefetching related objects into a local relation_cache. This should give us a big performance boost. I have also introduced assigning related objects by their id instead of full instances. #10 

Prefetching is enabled by default, Assigning by id is disabled. Both can be configured through parameters to the `is_a` function.

I've made some small benchmarks with the blog testapp from the repo using real data. It shows a speedup of 1.8 using sqlite3, real DBMS (not in-memory) should have massively bigger speedups.  https://gist.github.com/pboehm/11099324

Hope that this solves these performance problems.
